sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -123,35 +178,136 @@ namespace Azure { namespace Core { namespace Http {
     /**
      * @brief When `true`, allows an invalid certificate authority.
      */
-    bool IgnoreUnknownCertificateAuthority = false;
+    bool IgnoreUnknownCertificateAuthority{false};
+
+    /**
+     * Proxy information.
+     */
+
+    /**
+     * @brief If True, enables the use of the system default proxy.
+     *
+     * @remarks Set this to "true" if you would like to use a local HTTP proxy like "Fiddler" to
+     * capture and analyze HTTP traffic.
+     *
+     * Set to "false" by default because it is not recommended to use a proxy for production and
+     * Fiddler's proxy interferes with the HTTP functional tests.
+     */
+    bool EnableSystemDefaultProxy{false};
+
+    /**
+     * @brief If True, enables checks for certificate revocation.
+     */
+    bool EnableCertificateRevocationListCheck{false};
+
+    /**
+     * @brief Proxy information.
+     *
+     * @remark The Proxy Information string is composed of a set of elements
+     * formatted as follows:
+     * ([<scheme>=][<scheme>"://"]<server>[":"<port>])
+     *
+     * Each element should be separated with semicolons or whitespace.
+     */
+    std::string ProxyInformation;
+
+    /**
+     * @brief User name for proxy authentication.
+     */
+    std::string ProxyUserName;
```

Comment:
Similar to curl should this be Nullable to allow three states (Null, Empty, Something).

Reply:
What are the semantics for `ProxyUserName` being empty as opposed to nullable? What does it mean for it to be empty?

I came up with extremely reasonable semantics for `ProxyPassword` and `HttpProxy` (empty `ProxyPassword` means an empty password, empty `HttpProxy` means disable the use of a proxy server).

I couldn't come up with semantics for what an empty `ProxyUserName` meant, which is why it's not nullable.